Add Barbican adoption with Proteccio HSM integration documentation (#1058)
openstack-k8s-operators/data-plane-adoption: docs_user/modules/con_key-manager-service-support-for-crypto-plugins.adoc

= Key Manager service support for crypto plug-ins

[role="_abstract"]

> The Key Manager service (barbican) supports multiple crypto plug-ins for different security requirements and deployment scenarios.

Reviewer: ```suggestion
The {key_manager_first_ref} supports multiple crypto plug-ins for different security requirements and deployment scenarios.
```

> == Supported crypto plug-ins

Reviewer: ```suggestion
```

The following crypto plug-ins are supported in {rhos_acro}:

* **Simple Crypto Plugin**: Software-based cryptographic backend suitable for development and testing environments

> * **PKCS#11 Plugin**: Hardware Security Module (HSM) integration for production environments requiring enhanced security

Reviewer: ```suggestion
* Simple Crypto Plug-in: Software-based cryptographic back end suitable for development and testing environments
* Public Key Cryptography Standard  (PKCS) #11 Plug-in: Hardware security module (HSM) integration for production environments that requires enhanced security
```

//*TODO: Talk about Ceph Storage and Swift Storage nodes, HCI deployments,
//etc.*

For environments that require HSM integration during adoption, see xref:adopting-key-manager-service-with-proteccio-hsm_{context}[Adopting the {key_manager} with Proteccio HSM integration].

== HSM Integration

> 

Reviewer: ```suggestion
```

> The Key Manager service supports Hardware Security Module (HSM) integration through the PKCS#11 plugin. This enables:

Reviewer: ```suggestion
The {key_manager} supports HSM integration through the PKCS#11 plugin. This enables:
```

* Enhanced security for secret storage and retrieval
* Hardware-based key management
* Compliance with security requirements for production environments
* Preservation of existing HSM-protected secrets during adoption

> HSM integration is configured through a simple boolean flag (`barbican_hsm_enabled`) during the adoption process.

Reviewer: ```suggestion
HSM integration is configured through a simple boolean flag (`barbican_hsm_enabled`) during the adoption process. For environments that require HSM integration during adoption, see xref:adopting-key-manager-service-with-proteccio-hsm_{context}[Adopting the {key_manager} with Proteccio HSM integration].
```
